### DISCUSSION TITLE
@mentions in activity service - should only send notification to users that have app access

### DISCUSSION THREAD

**Discussion by A4TIF** [directus/directus#24874] [Feature Requests]
### Describe the Bug

When using mentions @<username>

![image](https://github.com/directus/directus/assets/7469885/c729e3ab-e3d0-4f8c-8b54-876202b57af7)

If I select a user that doesn't have access to the app, the user should not be receiving the notifications. Or better, the only users available for mentions should be the one with app access, or we should be able to remove the "Click here to view" link from the email template. 

This should be fixed in https://github.com/directus/directus/blob/main/api/src/services/activity.ts

Ideally, if we are allowing notifications, then I dont want users to see my db table names (as displayed in the subject line), ex: `You were mentioned in directus_users` and the "Click here to view" should be disabled for users without app access.

### To Reproduce

Add a comment on any collection, and add text with @<any username that doesnt have app access>, they will get a notification (with email) and in the template there's a link to access it. 

### Directus Version

v10.6.1

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by paescuj** [comment]
I think I agree that notifications should only be sent to users who have app access. Feels like a sensible and simple solution to me. WDYT @rijkvanzanten?

**Comment by A4TIF** [comment]
On second thought, I believe the notification idea is good as users may have email notifications enabled and if we are @mentioning them then let them see the context of it, but let's not send the URL to the object.
